### ISSUE TITLE
Document format for .csv files to upload to data.wikiwatershed.org?

### ISSUE THREAD

**Issue by dbressler75** [ODM2/ODM2DataSharingPortal#294]
Does the .csv file need to be in a certain format when uploading?  I'm assuming so but there is no direction on this.  

Also, is there a difference between submitting a .csv for a file that is not online versus a file that is online but has data gaps that are to be filled in via upload of .csv file?  

![envirodiy_uploadcsvfile_question](https://user-images.githubusercontent.com/39309124/45508089-e1a1fb00-b761-11e8-9cd6-a956b02ab665.JPG)

**Comment by horsburgh** [comment]
Per the request from @aufdenkampe - We implemented this so that the CSV that is uploaded should be the same format as the CSV that is output by the EnviroDIY datalogger.

That I know of, there isn't a difference between whether a site is online or not. @jcaraballo17 can let us know if there are any differences, but I believe that his code just scans through the uploaded file and tries to add the data to the database if it doesn't exist already.

**Comment by aufdenkampe** [comment]
@dbressler75, the format is what comes from the ModularSensors library.
See https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-403880885 and following comments for examples.

**Comment by jcaraballo17** [comment]
@dbressler75, it makes no difference whether the site is online or not. the file upload just inserts all the data points in the file if they are not already in the database.

**Comment by dbressler75** [comment]
I went through [#12 (comment)](https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-403880885) but I'm kind of lost in terms of finding an example of a correctly formatted csv file.  Could someone provide an example of a csv file that was successfully uploaded?  The formatting of the file can vary depending on the code on the specific mayfly board so I'm wondering about these types of issues.  Are there plans to provide step by step instructions or examples on doing this type of upload?  Maybe there are more issues to work out before users who are not prepared to figure out formatting themselves can use this feature(?) - no problem if so, I just have some folks who are offline and would like to be able to upload data.

**Comment by horsburgh** [comment]
@aufdenkampe and @SRGDamia1 - can you help @dbressler75 here? I just looked at issue #12, and there's so much discussion in that issue I'm sure it's hard to follow.

@dbressler75 - We designed the file uploader to use the file format that is output by the ModularSensors Arduino Library. So, if you are trying to upload file formats other than what is output by that library, they won't work. Our rationale for implementing it this way was that users should not have to do anything to format the files prior to uploading them, they just need to make sure that their site is correctly set up in the Data Sharing Portal. This assumes that you set up the site in the data sharing portal and added all of the variables you are measuring so that the unique identifiers associated with the site and variables show up correctly in the data file. The file uploader matches the data to the site and variable using these identifiers.

Also - there are some gotchas here - if you download one of the files from the logger and then open it in Excel and save it, a bunch of your original formatting will be changed. Excel tries to "help" you by changing the date/time format and removing text qualifiers, etc.

**Comment by aufdenkampe** [comment]
@dbressler75, we provide lots of examples in these specific comments:
* https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-405669275
* https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-405675388

However, as is noted by @horsburgh above, I discovered lots of issues when creating CSV files from Excel.  See my comment on Aug. 15 at https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-413281393.

@horsburgh and @jcaraballo17, can we use some of the nice functions in Pandas, such as the ability to parse nearly any `datetime` format, to make this all work much more smoothly?

@dbressler75, you are right that we need to document this, perhaps as we resolve #216 **Help: Create a help section**.

**Comment by horsburgh** [comment]
@aufdenkampe and @SRGDamia1 - I've assigned this issue to you. The upload functionality has been added to the site and so it looks like what is left on this issue is to add some documentation to the help pages to explain the format - and potentially provide some guidance around creating files.

With respect to your question about using Pandas, we plan to push a fix to the CSV download functionality with the next release. Rearchitecting to us Pandas is something that we will look at after that.

**Comment by jcaraballo17** [comment]
Taking this issue off the milestone. Maybe this should be another issue for adding content to the help section in https://wikiwatershed.org? 
Feel free to add it back if you think it belongs here.

**Comment by aufdenkampe** [comment]
@SRGDamia1, would you be able to work with @HeatherBrooks and @dbressler75 to post some documentation on the expected CSV file format at https://wikiwatershed.org/help/sensor-help/?

**Comment by SRGDamia1** [comment]
@dbressler75 - don't you and Scott have something?

**Comment by HeatherBrooks** [comment]
@SRGDamia1 @dbressler75 I just spoke with Scott Ensign and he pointed me to the EnviroDIY Mayfly Sensor Station Manual. If you can confirm that the instructions under "Option 2. Manual Upload" in [this section of the manual](https://www.envirodiy.org/mayfly-sensor-station-manual/#registering-a-station-on-monitormywatershedorg) are correct, I will add them to [https://wikiwatershed.org/help/sensor-help/](https://wikiwatershed.org/help/sensor-help/).

**Comment by SRGDamia1** [comment]
[Uploading CSVs to MMW.docx](https://github.com/ODM2/ODM2DataSharingPortal/files/3452758/Uploading.CSVs.to.MMW.docx)

**Comment by aufdenkampe** [comment]
@SRGDamia1, thanks for sharing these instructions.
I think it would be helpful to also include a few examples with the instructions, just including the headers and a few lines of data. Could you provide one or two to @HeatherBrooks?

**Comment by HeatherBrooks** [comment]
@aufdenkampe @SRGDamia1 @dbressler75 I've posted the instructions at [https://wikiwatershed.org/help/sensor-help/uploading-csv-sensor-data/](url).

**Comment by HeatherBrooks** [comment]
@dbressler75 Additional instructions and an example CSV file have been added to the [documentation](https://wikiwatershed.org/help/sensor-help/sharing-sensor-data/#manually-uploading-data-from-a-file).

**Comment by aufdenkampe** [comment]
This should have been closed quite a while ago!
